myhpi/core/templatetags/core_extras.py

```diff
@@ -3,6 +3,42 @@
 register = template.Library()
 
 
+@register.inclusion_tag("menu_level.html")
+def build_menu_level(sub_pages_by_level_and_id, level):
+    return {"sub_pages_by_id": sub_pages_by_level_and_id.get(level, {}), "level": level}
+
+
+@register.filter
```

Comment:
Just to understand what is happening: why is `@register.filter` occurring 3 times here?

Reply:
In my understanding, this is an annotation of the following function. Functions annoted with `@register.filter` can be used in HTML files